bitpoll/poll/spam_util.py

```
@@ -0,0 +1,66 @@
+import random
+import time
+from typing import Dict, Tuple
+
+from django.conf import settings as django_settings
+from django.core import signing
+from django.core.signing import BadSignature
+from django.utils.translation import ugettext_lazy as _
+
+
+def create_anti_spam_challenge(poll_id: int) -> Dict:
+    """Creates a anti spam challenge"""
+    x = random.randint(1, 10)
+    y = random.randint(1, 10)
+    op = random.choice(['+', '-', '*'])
+    return {'key': signing.dumps({
+                'x': x,
+                'y': y,
+                'op': op,
+                'poll_id': poll_id,
+                'time': time.time()
+            }),
+        'x': x,
+        'y': y,
+        'op': op,
+    }
+
+
+def get_spam_challenge_from_key(key: str, poll_id: int) -> Dict:
+    """Returns the anti spam challenge from the key, or generates a new if it is not valid"""
+    try:
+        spam_data = signing.loads(key)
+        spam_data['key'] = key
+        if spam_data['time'] <= time.time() - django_settings.ANTI_SPAM_CHALLENGE_TTL:
+            return create_anti_spam_challenge(poll_id)
+        return spam_data
+    except BadSignature:
+        return create_anti_spam_challenge(poll_id)
+
+
+def check_anti_spam_challange(key: str, answer: str, poll_id: int) -> Tuple[bool, str]:
```

I think I would rather prefer the validation procedure of Django forms where a `ValidationError` (or so) is raised on error that contains a `message` attribute rather than having this interesting return signature.
(Feel free to ignore this comment)